openproblems/tasks/_batch_integration/batch_integration_graph/methods/baseline.py

```diff
@@ -27,43 +26,22 @@ def no_integration(adata, test=False):
     return adata
 
 
-def _randomize_subgraph(distances, connectivities):
-    row = np.random.choice(
-        np.arange(distances.shape[0]), len(distances.data), replace=True
-    )
-    col = np.random.choice(
-        np.arange(distances.shape[0]), len(distances.data), replace=True
+def _randomize_graph(adata, partition=None):
+    distances, connectivities = (
+        adata.obsp["uni_distances"],
+        adata.obsp["uni_connectivities"],
     )
-    while np.any(np.any(row == col)):
-        # eliminate self-loops
-        row[row == col] = np.random.choice(
-            np.arange(distances.shape[0]), np.sum(row == col)
-        )
-    # eliminate duplicates
-    row, col = zip(*set(zip(row, col)))
-    distances_data = np.random.permutation(distances.data)[: len(row)]
-    distances_out = scipy.sparse.coo_matrix(
-        (distances_data, (row, col)), shape=distances.shape
-    ).tocsr()
-    connectivities_data = np.random.permutation(connectivities.data)[: len(row)]
-    connectivities_out = scipy.sparse.coo_matrix(
-        (connectivities_data, (row, col)), shape=connectivities.shape
-    ).tocsr()
-    return distances_out, connectivities_out
-
-
-def _randomize_graph(distances, connectivities, batch=None):
-    if batch is None:
-        return _randomize_subgraph(distances, connectivities)
+    if partition is None:
+        new_idx = np.random.permutation(np.arange(distances.shape[0]))
     else:
-        distances_out = scipy.sparse.csr_matrix(distances.shape)
-        connectivities_out = scipy.sparse.csr_matrix(connectivities.shape)
-        for batch_name in np.unique(batch):
-            idx = np.argwhere(batch == batch_name).flatten()
-            distances_out[idx], connectivities_out[idx] = _randomize_subgraph(
-                distances[idx], connectivities[idx]
-            )
-        return distances_out, connectivities_out
+        new_idx = np.arange(distances.shape[0])
+        for partition_name in np.unique(partition):
+            partition_idx = np.argwhere(partition == partition_name).flatten()
+            new_idx[partition_idx] = np.random.permutation(new_idx[partition_idx])
```

Comment:
this looks like you could just do `_randomize_features(np.arange(distances.shape[0], partition_labels)` instead of rewriting this function, no?

Reply:
You're right :)

Reply:
just need the wrapper for `adata.obsp` `distances` and `connectivities` below then.